interfaces/builtin/overmount.go

```diff
@@ -0,0 +1,156 @@
+	return "overmount"
+}
+
+func (iface *OvermountInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface %q", iface))
+	}
+
+	// check that we have either a read or write path
+	rpath := iface.path(slot, "read")
+	wpath := iface.path(slot, "write")
+	epath := iface.path(slot, "execute")
+	if len(rpath) == 0 && len(wpath) == 0 && len(epath) == 0 {
+		return fmt.Errorf("read, write or execute path must be set")
+	}
+
+	// go over both paths
+	paths := rpath
+	paths = append(paths, wpath...)
+	for _, p := range paths {
+		if !cleanSubPath(p) {
+			return fmt.Errorf("overmount interface path is not clean: %q", p)
+		}
+	}
+
+	return nil
+}
+
+func (iface *OvermountInterface) SanitizePlug(plug *interfaces.Plug) error {
+	if iface.Name() != plug.Interface {
+		panic(fmt.Sprintf("plug is not of interface %q", iface))
+	}
+	target, ok := plug.Attrs["target"].(string)
+	if !ok || len(target) == 0 {
+		return fmt.Errorf("overmount plug must contain target path")
+	}
+	if !cleanSubPath(target) {
+		return fmt.Errorf("overmount interface target path is not clean: %q", target)
+	}
+
+	return nil
+}
+
+func (iface *OvermountInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *OvermountInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+// path is an internal helper that extract the "read" and "write" attribute
+// of the slot
+func (iface *OvermountInterface) path(slot *interfaces.Slot, name string) []string {
+	if name != "read" && name != "write" && name != "execute" {
+		panic("internal error, path can only be used with read/write/execute")
+	}
+
+	paths, ok := slot.Attrs[name].([]interface{})
+	if !ok {
+		return nil
+	}
+
+	out := make([]string, len(paths))
+	for i, p := range paths {
+		out[i], ok = p.(string)
+		if !ok {
+			return nil
+		}
+	}
+	return out
+}
+
+func (iface *OvermountInterface) overmountMountSnippet(plug *interfaces.Plug, slot *interfaces.Slot) ([]byte, error) {
+	snippet := bytes.NewBuffer(nil)
+	for _, r := range iface.path(slot, "read") {
+		fmt.Fprintln(snippet, mountEntry(plug, slot, r, ",ro"))
+	}
+	for _, w := range iface.path(slot, "write") {
+		fmt.Fprintln(snippet, mountEntry(plug, slot, w, ""))
+	}
+	for _, e := range iface.path(slot, "execute") {
+		fmt.Fprintln(snippet, mountEntry(plug, slot, e, ",exec"))
+	}
+	return snippet.Bytes(), nil
+}
+
+func appArmorEntry(plug *interfaces.Plug, slot *interfaces.Slot, relSrc, mntOpts, permissions string) string {
+	/* dst */ _ = resolveSpecialVariable(plug.Attrs["target"].(string), plug.Snap)
+	/* src */ _ = resolveSpecialVariable(relSrc, slot.Snap)
+	return fmt.Sprintf("mount options=(ro bind %s) %s/*/** -> %s/** %s,", "/snap", "exec", "/usr/bin", permissions)
```

Comment:
The interface must actually work. The hardcoding here doesn't make sense for the actual implementation.